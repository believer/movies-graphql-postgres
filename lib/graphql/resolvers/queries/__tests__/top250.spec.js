const top250 = require('../top250')

describe('queries/top250', () => {
  let db

  beforeEach(() => {
    db = {
      query: jest.fn().mockReturnValue('dbData'),
    }
  })

  it('should query for top250', () => {
    top250(null, { limit: 200, offset: 0 }, { db })

    expect(db.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('should return database data', async () => {
    const response = await top250(null, { limit: 200, offset: 0 }, { db })

    expect(response).toEqual('dbData')
  })

  it('should handle errors', async () => {
    db.query.mockImplementation(() => Promise.reject('nope'))

    try {
      await top250(null, { limit: 200, offset: 0 }, { db })
    } catch (e) {
      expect(e.message).toMatch(/nope/)
    }
  })
})
