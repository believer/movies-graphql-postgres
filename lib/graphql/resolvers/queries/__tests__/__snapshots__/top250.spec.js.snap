// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries/top250 should query for top250 1`] = `
"
    SELECT 
      avg(r.rating) as average_rating, 
      count(r.rating) as number_of_ratings, 
      m.*,
      EXTRACT(YEAR FROM m.release_date) as year
    FROM ratings AS r 
    INNER JOIN movies AS m ON m.id = r.movie_id 
    GROUP BY m.id 
    ORDER BY
      number_of_ratings DESC,
      average_rating DESC,
      title ASC
    OFFSET 0
    LIMIT 200
  "
`;
